src/orquestra/sdk/_base/_dsl.py

```diff
@@ -397,6 +397,7 @@ def __init__(
         resources: Resources = Resources(),
         custom_image: Optional[str] = None,
         custom_name: Optional[str] = None,
+        custom_source_import: Optional[bool] = True,
```

Comment:
Have you considered making `source_import` optional instead of adding a separate flag? 

Why I think it would be better: `custom_source_import` has a similar name to `custom_image` and `custom_name` but the `*_import`'s semantics is different; it's a flag that determines if the `source_import` should be used or ignored. Seems like a use case for optionals.